addons/project/controllers/project_sharing_chatter.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug.exceptions import Forbidden
+
+from odoo.http import request, route
+from odoo.tools import consteq
+
+from odoo.addons.portal.controllers.mail import PortalChatter
+from .portal import ProjectCustomerPortal
+
+
+class ProjectSharingChatter(PortalChatter):
+    def _check_project_sharing_access(self, project_id, res_model, res_id, token):
```

Comment:
it's a bit unfortunate that this helper method has the same name as the one it calls on `project.project`, especially as they have different roles and signatures. Maybe rename it to `_check_project_access_and_get_token()` or something that shows the difference, and the fact that it returns a token :thinking: